NitroxPatcher/Patches/Dynamic/Player_SetCurrentSub_Patch.cs

```diff
@@ -14,16 +14,20 @@ public class Player_SetCurrentSub_Patch : NitroxPatch, IDynamicPatch
         public static readonly Type TARGET_CLASS = typeof(Player);
         public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("SetCurrentSub", BindingFlags.Public | BindingFlags.Instance);
 
-        public static void Prefix(SubRoot sub)
+        public static void Prefix(Player __instance, SubRoot sub)
         {
             NitroxId subId = null;
 
             if (sub != null)
             {
                 subId = NitroxEntity.GetId(sub.gameObject);
             }
-
-            NitroxServiceLocator.LocateService<LocalPlayer>().BroadcastSubrootChange(Optional.OfNullable(subId));
+            // When in the water of the moonpool, it can happen that you hammer change requests
+            // while the sub is not changed. This will prevent that
+            if (__instance.GetCurrentSub() != sub)
```

Comment:
Is this the fix?  I thought in the moonpool, the player is being forced out of the sub with waves?  Per this, the subroot is actually staying the same?

Reply:
Yes the subroot actually does not change. But the client calls the subrootchanged method anyway (like a machinegun)